File: 04.BackendRTE/src/main/java/de/montigem/be/dtos/rte/IdDTO.java
/*
 *  (c) Monticore license: https://github.com/MontiCore/monticore
 */

package de.montigem.be.dtos.rte;

public class IdDTO extends DTO {

  protected IdDTO() {
    super("IdDTO");
  }

  public IdDTO(long id) {
    this();
    this.id = id;
  }
}

File: 03.FrontendRTE/src/app/shared/architecture/command/aggregate/id.dto.ts
/* (c) https://github.com/MontiCore/monticore */

import { JsonMember, JsonObject } from '@upe/typedjson';
import { DTO } from '@shared/architecture';

@JsonObject()
export class IdDTO extends DTO<IdDTO> {
  @JsonMember({
    isRequired: true,
    name: 'id',
    type: Number,
  }) private _objectId: number;

  constructor() {
    super();
    this.typeName = 'IdDTO';
    this._objectId = 0;
  }

  get objectId(): number {
    return this._objectId;
  }

  set objectId(value: number) {
    this._objectId = value;
  }
}
